### PR TITLE
Salvage filter generic events

### PR DESCRIPTION
**Pull request by lptr**
Salvage some of the good stuff from abandoned PR https://github.com/gradle/gradle/pull/14783.

### CODE REVIEW

**Comment by lptr** [comment]
@bot-gradle test QFL please.

File: subprojects/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApiExecution.groovy [hunk 1]
```diff
@@ -38,7 +38,7 @@ class ToolingApiExecution extends AbstractMultiTestRunner.Execution implements T
 
     @Override
     protected String getDisplayName() {
-        return "${displayName(toolingApi.version)} -> ${displayName(gradle.version)}"
+        return "TAPI: ${displayName(toolingApi.version)}, target: ${displayName(gradle.version)}"
```
Comment (wolfs) [inline subprojects/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApiExecution.groovy:41]: I am not sure this makes things easier to understand. Maybe `TAPI` and `Gradle` would be better. Or something like `TAPI 6.7 -> Gradle 4.5`. Do you think it makes sense to get some feedback on this rename on Slack?

Comment (lptr) [reply subprojects/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApiExecution.groovy:41]: @donat what do you think?

Comment (donat) [reply subprojects/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApiExecution.groovy:41]: I'm fine even with the original notation, but I guess I just got used to it. 

The `TAPI 6.7 -> Gradle 4.5` notation is a bit more explicit, I like it better than `target`.

Comment (lptr) [reply subprojects/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApiExecution.groovy]: `TAPI 6.7 -> Gradle 4.5` it is!

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Review by wolfs — APPROVED** [review]
LGTM!

**Pull request merged**